tests/gen/shift/6.toml

```diff
@@ -0,0 +1 @@
+params = [32, 6]
```

Comment:
Can we use a slightly more explicit syntax here:
```
main.params = [32, 6]
```

In the future, we might want multiple externally-visible components.

Reply:
Sure, but just as a note if we want this to be supported for multiple components we could no longer have the whole `[A, B]` param system and would have to build something different anyway.

Reply:
What do you mean?

Reply:
Because normal parameters run the risk of being double bound by both the invoking component and the configuration, so we would want to be able to instead specify parameters that are only modified by the configuration file (like in #441).